lib/grains_cases.rb

```diff
@@ -0,0 +1,49 @@
+class GrainsCase < OpenStruct
+  def test_name
+    'test_%s' % description.downcase.tr_s(' ', '_')
+  end
+
+  def workload
+    fail NotImplementedError
+  end
+
+  def skipped
+    index == 0 ? '# skip' : 'skip'
+  end
+
+  private
+
+  def formatted_expected
+    expected.to_s.reverse.gsub(/...(?=.)/,'\&_').reverse
```

Comment:
This would be nicer as a function that took an argument.
```
def underscore_format(number)
```
Then you use it as: `underscore_format(expected)`